Signal/src/network/GiphyDownloader.swift

```diff
@@ -46,15 +51,27 @@ class GiphyAssetSegment {
     }
 
     public func append(data: Data) {
+        guard state == .downloading else {
+            owsFail("\(TAG) appending data in invalid state: \(state)")
+            return
+        }
+
         datas.append(data)
     }
 
     public func mergeData(assetData: NSMutableData) {
+        guard state == .complete else {
```

Comment:
Not functionally wrong, but why are we using `NSMutableData`?  I'd expect something like:

```
mergeData(assetData: inout Data) {
...
assetData.append(data.subdata(in: start..<end))
...
}

var assetData = Data()
mergeData(assetData: &assetData)
```

Reply:
Done.